### ISSUE TITLE
AS4RequestHandler does not use correct BDEW profile crypto and signing params

### ISSUE THREAD

**Issue by problemzebra2** [phax/phase4#166]
`AS4RequestHandler` for handling incoming AS 4 messages seems to construct new instances of `AS4SigningParams` and `AS4CryptParams`. The required parameters according to the official BDEW profile are only set in `Phase4BDEWSender.AbstractBDEWUserMessageBuilder`

```
// Other crypt parameters are located in the PMode security part
cryptParams ().setKeyIdentifierType (DEFAULT_KEY_IDENTIFIER_TYPE);
cryptParams ().setKeyEncAlgorithm (ECryptoKeyEncryptionAlgorithm.ECDH_ES_KEYWRAP_AES_128);
cryptParams ().setEncryptSymmetricSessionKey (false);

/**
 * Assumption: the BST "ValueType" attribute is set to
 * "http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-x509-token-profile-1.0#X509PKIPathv1"
 * by WSS4J automatically (see WSSecSignature#addBST)
 */

// See BDEW specs 2.2.6.2
// Other signing parameters are located in the PMode security part
signingParams ().setKeyIdentifierType (DEFAULT_KEY_IDENTIFIER_TYPE);
signingParams ().setAlgorithmC14N (ECryptoAlgorithmC14N.C14N_EXCL_OMIT_COMMENTS);
// Use the BST value type "#X509PKIPathv1"
signingParams ().setUseSingleCertificate (false);
```

Is this a bug or is there a way to specify the required crypto and signing params for the `AS4RequestHandler` used for incoming AS 4 messages?

**Comment by phax** [comment]
That seems to be very related to #165 isn't it?

**Comment by problemzebra2** [comment]
Yes, this seems to be a more general issue. #165 refers to a specific signing setting.

**Comment by phax** [comment]
I am trying to get a hold of this - I totally see the issue here

**Comment by problemzebra2** [comment]
By the way `cryptParams ().setKeyIdentifierType (DEFAULT_KEY_IDENTIFIER_TYPE);` seems to be wrong. According to the BDEW profile SKI_KEY_IDENTIFIER should be used. I will make a own issue for this.

**Comment by phax** [comment]
Well, chapter "2.2.6.2.1 Signatur" seem to clearly state BST:
![grafik](https://github.com/phax/phase4/assets/196219/9ca8ce6b-b259-464f-a617-4ba63260ca4c)

**Comment by phax** [comment]
The resolution is implemented in https://github.com/phax/phase4/commit/a02ebf6867b50f757e965596f0d29abcb83547ab
It changes `IAS4IncomingSecurityConfiguration` to include the full `AS4SigningParams` and `AS4CryptParams` objects.

**Comment by phax** [comment]
This should be resolved with the 2.3.0 release
